subprojects/platform-jvm/src/main/java/org/gradle/jvm/toolchain/internal/install/JdkCacheDirectory.java

```diff
@@ -160,6 +161,32 @@ public File provisionFromArchive(JavaToolchainSpec spec, File jdkArchive, URI ur
         }
     }
 
+    private void checkInstallFolderForLeftoverContent(File installFolder, URI uri, JavaToolchainSpec spec, JvmInstallationMetadata metadata) {
+        if (!installFolder.exists()) {
+            return; //install folder doesn't even exist
+        }
+
+        File[] filesInInstallFolder = installFolder.listFiles();
+        if (filesInInstallFolder == null || filesInInstallFolder.length == 0) {
+            return; //no files in install folder
+        }
+
+        File markerLocation = markedLocation(installFolder);
+        if (!isMarkedLocation(markerLocation)) {
+            return; //no marker found
+        }
+
+        String leftoverMetadata;
+        try {
+            leftoverMetadata = getMetadata(markerLocation).toString();
+        } catch (Exception e) {
+            leftoverMetadata = "Could not be determined due to: " + e.getMessage();
```

Comment:
Can we log the exception at `debug` level, just to have the stacktrace kept somewhere?

Reply:
Yes, definitely.